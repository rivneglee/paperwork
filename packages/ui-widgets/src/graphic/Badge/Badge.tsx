import React, { FunctionComponent, ReactElement } from 'react';
import classNames from 'classnames';

interface Props {
  className?: string;
  right?: ReactElement;
  left?: ReactElement;
  color?: 'primary' | 'secondary';
}

const Badge: FunctionComponent<Props> = ({
  children,
  className,
  color,
  left,
  right,
}) => (
  <div className={
    classNames(
      'pw-badge',
      color && `pw-badge--color-${color}`,
      className,
    )
  }>
    { left && left }
    <span className="pw-badge__label">{children}</span>
    { right && right }
  </div>
);

export default Badge;
