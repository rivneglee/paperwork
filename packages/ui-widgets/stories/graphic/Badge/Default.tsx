import React from 'react';

import { Badge, Card } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header={<h3>Badge colours</h3>}>
      <Badge>default</Badge>
      <Badge color="primary">primary</Badge>
      <Badge color="secondary">secondary</Badge>
    </Card>
  </div>
);
