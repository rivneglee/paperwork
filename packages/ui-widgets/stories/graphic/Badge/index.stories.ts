import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Badge', module)
  .add('default', Default);
