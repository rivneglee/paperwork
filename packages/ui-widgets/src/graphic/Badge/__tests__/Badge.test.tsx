import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Badge from '../Badge';

describe('Badge', () => {
  it('should render default spinner', () => {
    const wrapper: ShallowWrapper = shallow(
      <Badge>default</Badge>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with colours', () => {
    const wrapper: ShallowWrapper = shallow(
      <Badge color="primary">default</Badge>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with left and right element', () => {
    const wrapper: ShallowWrapper = shallow(
      <Badge left={<>left</>} right={<>right</>}>default</Badge>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
