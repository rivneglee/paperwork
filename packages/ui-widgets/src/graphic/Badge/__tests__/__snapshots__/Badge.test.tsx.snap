// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render default spinner 1`] = `
<div
  className="pw-badge"
>
  <span
    className="pw-badge__label"
  >
    default
  </span>
</div>
`;

exports[`Badge should render with colours 1`] = `
<div
  className="pw-badge pw-badge--color-primary"
>
  <span
    className="pw-badge__label"
  >
    default
  </span>
</div>
`;

exports[`Badge should render with left and right element 1`] = `
<div
  className="pw-badge"
>
  left
  <span
    className="pw-badge__label"
  >
    default
  </span>
  right
</div>
`;
